x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -206,8 +210,50 @@ else if (node instanceof OutputExec outputExec) {
     }
 
     private PhysicalOperation planAggregation(AggregateExec aggregate, LocalExecutionPlannerContext context) {
-        var source = plan(aggregate.child(), context);
-        return physicalOperationProviders.groupingPhysicalOperation(aggregate, source, context);
+        var child = aggregate.child();
+        PhysicalOperation operation;
+        // optimize Lucene count
+        if (child instanceof EsQueryExec queryExec) {
+            var target = nonGroupedCountTarget(aggregate);
+            if (target != null) {
+                return luceneCount(context, target instanceof String ? target.toString() : null);
+            }
+        }
+
+        var source = plan(child, context);
+        operation = physicalOperationProviders.groupingPhysicalOperation(aggregate, source, context);
+        return operation;
+    }
+
+    /**
+     * Returns the target count - null if it doesn't apply, Boolean.TRUE for count all or a String representing the underlying field name.
+     *
+     * TODO: need to handle runtime fields
+     */
+    private Object nonGroupedCountTarget(AggregateExec aggregate) {
+        if (aggregate.groupings().isEmpty()) {
+            var aggs = aggregate.aggregates();
+            if (aggs.size() == 1) {
+                var agg = aggs.get(0);
+                if (agg instanceof Alias as && as.child() instanceof Count count) {
+                    var expression = count.field();
+                    // if the count points to an actual field, push it down
+                    if (expression instanceof FieldAttribute fa) {
+                        return fa.name();
```

Comment:
I don't think we can always push down the count for a field  even it's not a runtime field.